RecipeMatch/View Controllers/SearchViewController.m

```diff
@@ -67,9 +69,10 @@ - (void)reloadSearch:(NSTimer *)timer {
             self.recipes = recipes;
             // only reload section 1 to prevent search bar from losing first responder status
             [self.collectionView reloadSections:[NSIndexSet indexSetWithIndex:1]];
-            
+        } else if (recipes) {
+            [[APIManager shared] cancelDataTask];
```

Comment:
because your `APIManager ` is a singleton that generates new `NSURLSessionDataTask`  for every request, this approach might lead to a case where you will cancel another (newer) request that was sent, and not necessary the one related to the search api.
so it's better have the searchApi (and the `requestFromAPIWithURL`) to return `NSURLSessionDataTask` back to this VC, and then cancel it locally